add focus search bar, open file, dataset shortcut

NakulManchanda opened the pull request:
Signed-off-by: Nakul Manchanda <nakul.manchanda@ibm.com>

NakulManchanda commented:
This PR contains following
1) added three new shortcuts
     a) Alt+O - open directory
     b) Alt+K - open dataset
     c) Alt+S - bring focus to serach bar

2) Also, listed shortcuts alongside menu-item

NakulManchanda commented:
added one more shortcut 
d) Alt+1 - to bring back the focus to editor

DivergentEuropeans commented on webClient/src/app/core/menu-bar/menu-bar.config.ts:
```diff
@@ -142,7 +142,8 @@ export const MENU = [
                 internalName: 'createFile'
               },
-              keyMap: '' // [Ctrl+N] won't work due to browser conflicts. Keybindings will need
+              keyMap: '', // [Ctrl+N] won't work due to browser conflicts. Keybindings will need
```
You can remove the [Ctrl + N] comment, no longer relevant. A new comment up at the top would be good, something like "Cross browser research shows ALT + [ ] are the least conflicting combinations, so we use these"

NakulManchanda replied:
removed old comments, also moved shortcut attribute to existing attribute keymap

DivergentEuropeans commented on webClient/src/app/core/menu-bar/menu-bar.config.ts:
```diff
@@ -173,8 +176,9 @@ export const MENU = [
                 },
-                keyMap: '' // [Ctrl+S] won't work due to browser conflicts. Keybindings will need
-            },              // to be rethinked.
+                keyMap: '', // [Ctrl+S] won't work due to browser conflicts. Keybindings will need
```
Is this still relevant. Does Ctrl + S conflict with things? Can we just be brave and use Alt + S, is it worth it? @NakulManchanda @1000TurquoisePogs

DivergentEuropeans replied:
Eh, Ctrl + S is too common, I don't even like this suggestion.

DivergentEuropeans commented on webClient/src/app/core/menu-bar/menu-bar.config.ts:
```diff
@@ -157,13 +158,15 @@ export const MENU = [
                 name: 'Open Directory',
```
@NakulManchanda  Additionally, now that we have a few different shortcuts, yet no documentation yet... can you please add the corresponding hotkey to the title of the action? Similar to this,
![image](https://user-images.githubusercontent.com/20528015/66782931-d6f5f000-eea4-11e9-95d1-2225ed63691a.png)

This was customly requested by our UX team as well.

DivergentEuropeans replied:
Oh sorry, you said you did this.

NakulManchanda commented:
> Hi Nakul, can you revisit the CSS to determine what you want to do here?
> I noticed the bullet for showing the current language now is on its own line due to CSS changes:
> ![image](https://user-images.githubusercontent.com/30730276/65847348-341f6c80-e30f-11e9-827f-9241f82c528c.png)

fixed this css bug, re-organized menu row

NakulManchanda commented:
@DivergentEuropeans 
- Small menu bar elements are clickable & selectable
  removing tabindex from html is more difficult that i thought, null does not work it just makes it 0 which is worse than -1 

- DOM tree in browser
   **How to check on this?**

- Add hotkey to "New Directory" 
   Just a suggestion - i believe we should remove "New Directory" as menu option, learning from vscode
Also, now we have a legit alternate, right click and Create New Directory
Let me know, if you agree - we can remove this from menu

- hotkey vs tab order 
 search bar find it quiet handy, if we want to really make it keyboard friendly, i know right now we do lost control of hotkeys and we had to click to gain focus and key listeners to work.

NakulManchanda commented:
@DivergentEuropeans 
DOM tree in browser
ok thanks for pointing out in chat - investigating why its blinking on "group lines" menu element

NakulManchanda commented:
> * Small menu bar elements are clickable & selectable
 - Will deal this in DOM tree fixes, need hook to adjust styles after they are added to DOM, in my opinion this will have same fix as DOM item below.
> * DOM tree in browser suggests some kind of poor Angular element resource management/recursion happening
 - this is not introduced with this PR
> * Add hotkey to "New Directory"
- Removed this from menu

- Rebased again to remove conflicts
- added Alt+ delete as delete shortcut

Looks good to merge. Let me know if you think otherwise. thanks!

NakulManchanda commented:
@DivergentEuropeans Agreed
 if we add delete for current selection -  "Delete" should be enough, but it should be part of file explorer repo i guess, 
 so far i have not added any keyboard shortcut on tree, but i have a local branch, will experiment and add that as separate PR in explorer project

Few others required for tree - Delete, Up, Down, Enter to open folder etc

NakulManchanda commented:
removed alt+del will replace by just "Delete" on tree in a separate PR

NakulManchanda commented:
@DivergentEuropeans thanks for the clue to fix  menu bar group end highlighting issue.
Added ng-if on that div, hope its good thanks!
Thanks again for thorough review!
